File: app/src/main/java/com/example/reminders20/di/modules/DBModule.java
package com.example.reminders20.di.modules;

import androidx.annotation.NonNull;
import androidx.room.Room;

import com.example.reminders20.RemindersApplication;
import com.example.reminders20.db.ReminderDao;
import com.example.reminders20.db.ReminderDatabase;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class DBModule {

    public static final String REMINDER_DB_NAME = "reminder_db";

    @Provides
    @Singleton
    ReminderDatabase provideReminderDatabase(RemindersApplication application) {
        return Room.databaseBuilder(application, ReminderDatabase.class, REMINDER_DB_NAME).build();
    }

    @Provides
    @Singleton
    ReminderDao provideReminderDao(@NonNull ReminderDatabase db) {
        return db.reminderDao();
    }
}

File: app/src/main/java/com/example/reminders20/di/modules/DBModule.kt
package com.example.reminders20.di.modules

import androidx.room.Room
import com.example.reminders20.RemindersApplication
import com.example.reminders20.db.ReminderDao
import com.example.reminders20.db.ReminderDatabase
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class DBModule {
    @Provides
    @Singleton
    fun provideReminderDatabase(application: RemindersApplication?): ReminderDatabase {
        return Room.databaseBuilder(application!!, ReminderDatabase::class.java, REMINDER_DB_NAME).build()
    }

    @Provides
    @Singleton
    fun provideReminderDao(db: ReminderDatabase): ReminderDao? {
        return db.reminderDao()
    }

    companion object {
        const val REMINDER_DB_NAME = "reminder_db"
    }
}